bee-hive/bee_hive/workflow.py

```diff
@@ -9,37 +9,38 @@
 import dotenv
 
 dotenv.load_dotenv()
+from agent import Agent
 
 class Workflow:
     agents = []
-    def __init__(self, agent_defs)
+    workflow = {}
+    def __init__(self, agent_defs, workflow):
         """Execute sequential workflow.
         input:
             agents: array of agent definitions
+            workflow: workflow definition
         """
         for agent_def in agent_defs:
-            agents.append(new Agent(agent_def))
-
-
-    def sequential_workflow(workflow):
-        """Execute sequential workflow.
-        input: workflow
-            workflow["spec"]["template"]["prompt"]: prompt for the agent
-            workflow["spec"]["template"]["agents"]: array of agent instance
-        """
-    
-        prompt = workflow["spec"]["template"]["prompt"]
-        agents = workflow["spec"]["template"]["agents"]
-        if (
-            workflow_yaml["spec"]["strategy"]["output"]
-            and workflow_yaml["spec"]["strategy"]["output"] == "verbose"
-           ):
-            run_workflow = run_streaming_agent
+            self.agents.append(Agent(agent_def))
+        self.workflow = workflow
+
+
+    def run(self):
+        """Execute workflow."""
+
+        if (self.workflow["spec"]["strategy"]["type"]  == "sequence"):
```

Comment:
would it make sense to pull this into a hidden helper function? something like:

```python
class Workflow:
    def run(self):
        if strategy == sequence:
            return _run_sequence(workflow)

    def _run_sequence(workflow):
        ...
```

Reply:
I agree. I was going to do when other types are added.  I do it now.